src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/FormatterServices.cs

```diff
@@ -0,0 +1,299 @@
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Globalization;
+using System.Diagnostics;
+using System.Runtime.Serialization.Formatters;
+
+namespace System.Runtime.Serialization
+{
+    public static class FormatterServices
+    {
+        private static readonly ConcurrentDictionary<MemberHolder, MemberInfo[]> s_memberInfoTable = new ConcurrentDictionary<MemberHolder, MemberInfo[]>();
+
+        private static FieldInfo[] GetSerializableFields(Type type)
+        {
+            if (type.GetTypeInfo().IsInterface)
+            {
+                return Array.Empty<FieldInfo>();
+            }
+
+            if (!type.GetTypeInfo().IsSerializable)
+            {
+                throw new SerializationException(SR.Format(SR.Serialization_NonSerType, type.GetTypeInfo().FullName, type.GetTypeInfo().Assembly.FullName));
+            }
+
+            var results = new List<FieldInfo>();
+            for (Type t = type; t != typeof(object); t = t.GetTypeInfo().BaseType)
+            {
+                foreach (FieldInfo field in t.GetFields(BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly))
+                {
+                    if ((field.Attributes & FieldAttributes.NotSerialized) != FieldAttributes.NotSerialized)
+                    {
+                        results.Add(field);
+                    }
+                }
+            }
+            return results.ToArray();
+        }
+
+        public static MemberInfo[] GetSerializableMembers(Type type)
+        {
+            return GetSerializableMembers(type, new StreamingContext(StreamingContextStates.All));
+        }
+
+        public static MemberInfo[] GetSerializableMembers(Type type, StreamingContext context)
+        {
+            if (type == null)
+            {
+                throw new ArgumentNullException(nameof(type));
+            }
+
+            // If we've already gathered the members for this type, just return them.
+            // Otherwise, get them and add them.
+            var mh = new MemberHolder(type, context);
+            MemberInfo[] members;
+            if (!s_memberInfoTable.TryGetValue(mh, out members))
+            {
+                members = GetSerializableFields(type);
+                s_memberInfoTable.TryAdd(mh, members);
+            }
+            return members;
+        }
+
+        public static void CheckTypeSecurity(Type t, TypeFilterLevel securityLevel)
+        {
+            // nop
+        }
+
+        // TODO #8133: Fix this to avoid reflection
+        private static readonly Func<Type, object> s_getUninitializedObjectDelegate = (Func<Type, object>)
+            typeof(string).GetTypeInfo().Assembly.GetType("System.Runtime.Serialization.FormatterServices")
+            ?.GetMethod("GetUninitializedObject", BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Static)
+            ?.CreateDelegate(typeof(Func<Type, object>));
```

Comment:
> I assume that we switch this to call  System.Runtime.CompilerServices.RuntimeHelpers.GetUninitializedObject  to fix the reflection TODO.

Yes, that's what I had in mind.  Or, rather, I'd planned to go add a SerializationAugments ala the EnvironmentAugments we added, but in the process of doing so noticed you already added the RuntimeHelpers method.  So we can just use that.  I've postponed it as we'll need to change how this library builds to reference the platform components, at which point it should be a very easy change.

> Do we want to be calling this method via reflection as well for now?

You mean instead of creating the delegate?  Seemed like the delegate approach would provide for faster execution until we can remove the reflection.  Is that not the case?  I don't have a strong preference, though.  This reflection stuff shouldn't be here very long.